fml/message_loop.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/message_loop.h"
+#include <utility>
+#include "flutter/fml/message_loop_impl.h"
+#include "flutter/fml/task_runner.h"
+#include "flutter/fml/thread_local.h"
+#include "lib/ftl/memory/ref_counted.h"
+#include "lib/ftl/memory/ref_ptr.h"
+
+namespace fml {
+
+FML_THREAD_LOCAL ThreadLocal TLSMessageLoop([](intptr_t value) {
+  delete reinterpret_cast<MessageLoop*>(value);
+});
+
+MessageLoop& MessageLoop::GetCurrent() {
+  auto loop = reinterpret_cast<MessageLoop*>(TLSMessageLoop.Get());
+  if (loop != nullptr) {
+    return *loop;
+  }
+
+  loop = new MessageLoop();
```

Comment:
That makes sense.  The thing I'm worried about is someone accidentally touching GetCurrent() on a thread that isn't supposed to have a message loop and thereby creating one by mistake.  Would it be reasonable to have a static `EnsureInitializedForCurrentThread` call that makes it explicit when we're trying to spin up a message loop?  The code in thread.cc would make this call early in thread initialization.